Automattic/woocommerce-payments
# Restrict PM to domestic country

Fixes #6517 

#### Changes proposed in this Pull Request

Add property to `UPE_Payment_Method` class indicating that currency should be restricted to Stripe's connected account currency. This change is required for newly introduced BNPL methods - Affirm and Affterpay/Clearpay. 

See associated issue for more details about the problem. 

#### Testing instructions

1. Make sure merchant in US based account and multicurrecy turned on with at least CAD as option
2. Enable Affirm and Afterpay 
3. Add at least 50 USD to cart (Affirm is not available below that amount)
4. Go to checkout, default currency should be USD and both methods available in UPE element
5. Switch currency to CAD
6. Payment method should not be available anymore.

If same steps are repeated without proposed changes, we observe error in step 6.


**Turn off multi-currency** 
- the solution shouldn't brake existing checkout
- we expect that PMs are not available if currency is outside merchants currency (for example, EUR merchant still doesn't have Affirm)


-------------------


**Post merge**
